### ISSUE TITLE
Support JSX / TSX

### ISSUE THREAD

**Issue by voxpelli** [neostandard/neostandard#11]
I'll look into how JSX is handled in ESLint 9 / flat configs then

I guess TSX should be supported as well then

_Originally posted by @voxpelli in https://github.com/neostandard/neostandard/issues/5#issuecomment-2054160791_

**Comment by radek-duchon** [comment]
I'm not sure how exactly I could help, but I would love to see this issue to be closed :wink:
I have managed to set up `neostandard` to work with `jsx` using such a config:
```javascript
'use strict'

const reactPlugin = require('eslint-plugin-react')

const neo = require('neostandard')({
  ts: true,
  ignores: [
    'lib',
  ],
})

module.exports = [
  ...neo,
  {
    files: ['**/*.{jsx,mjsx}'],
    ...reactPlugin.configs.flat.recommended,
  },
]

module.exports.plugins = /** @type {const} */ ({
  get '@stylistic' () {
    return require('@stylistic/eslint-plugin')
  },
  get n () {
    return require('eslint-plugin-n')
  },
  get promise () {
    // @ts-ignore
    return require('eslint-plugin-promise')
  },
  get 'typescript-eslint' () {
    return require('typescript-eslint')
  },
  get react () {
    return require('eslint-plugin-react')
  },
})
```
It is linting `jsx` files, applying `neostandard` rules, not complaining about JSX... so I'm more-less happy. But as I'm experimenting with `neostandard` (trying to migrate our repos from custom `eslint` or `standard`), I do not see possible issues with that in the future or more complex (mono)repos.

But I could not make it work with `tsx` (yet). 😢

**Comment by voxpelli** [comment]
@radek-duchon Excellent, that’s useful feedback, eslint-plugin-react got support for ESLint 9 the other day, so yeah, should start looking at this

Would you maybe be interested in making a PR for initial JSX support?

**Comment by radek-duchon** [comment]
OK. I have 0 knowledge of your code, but why not?! I did not think that you would be interested at all. While making any changes to `nonstandard` code, I might be successful in adding the support for TSX. Let me see... 
I already had to git-clone `eslint-plugin-react` and using `npm link` add it to my test repo (they did add the support to `master`, but did not release the version yet - I should have mentioned that before 👼 ), so I can play a bit with `neostandard` too.

**Comment by voxpelli** [comment]
@radek-duchon Always fun to help guide a new contributor as well 😁

**Comment by radek-duchon** [comment]
It's working! 🎉  I have both, JSX and TSX support. I just have to refactor and re-format the code and I will file a PR.

**Comment by radek-duchon** [comment]
I wanted to push my working version and create a draft PR to get some early feedback (I might not be able to refactor the code today), but I do not have push access rights. Should I fork?

**Comment by jerome-benoit** [comment]
> I wanted to push my working version and create a draft PR to get some early feedback (I might not be able to refactor the code today), but I do not have push access rights. Should I fork?

Always to submit code to repos you're not a member.

**Comment by voxpelli** [comment]
> Should I fork?

Yeah, like Jerome says, fork it and submit a PR from your fork :)

**Comment by radek-duchon** [comment]
I have rebased my PR and tried to address all comments. Is there anything else I could/should do? I was thinking that I could completely remove the ruleset option and always add only `recommended` rules from the plugin. To begin with something... We would love to have JSX support in our projects.

**Comment by voxpelli** [comment]
@radek-duchon I have been on somewhat of an unplanned vacation last week so haven’t had time to look yet, will give the PR a look soon

**Comment by radek-duchon** [comment]
I have figured out a more simple configuration that works with both JSX and TSX. With every change I do I understand your code more and more 🤷 Tomorrow I will address @jerome-benoit 's feedback and I will try to integrate (at least partially) this configuration into my PR... For anyone interested or anyone who desperately wants JSX/TSX to work with `neostandard`, here is the configuration:
```
'use strict'

const reactPlugin = require('eslint-plugin-react')
const neostandard = require('neostandard')

const ignores = [
  ...neostandard.resolveIgnoresFromGitignore(),
]

const settings = {
  react: {
    version: 'detect',
  },
}

module.exports = [
  ...neostandard({
    ignores,
    ts: true,
    filesTs: ['**/*.tsx'],
  }),
  {
    files: ['**/*.{jsx,tsx}'],
    ignores,
    name: 'react/recommended',
    ...{
      ...reactPlugin.configs.flat.recommended,
      settings,
    },
  },
  {
    files: ['**/*.{jsx,tsx}'],
    ignores,
    name: 'react/jsx-runtime',
    ...{
      ...reactPlugin.configs.flat['jsx-runtime'],
      settings,
    },
  },
]
```
Thanks for the support, both of you (@voxpelli & @jerome-benoit)

**Comment by voxpelli** [comment]
Released in `0.11.2` 🙏
